apps/builder/app/canvas/features/wrapper-component/wrapper-component.tsx

```diff
@@ -167,12 +169,14 @@ export const WrapperComponentDev = ({
           />
         }
         onChange={(updates) => {
-          store.createTransaction([rootInstanceContainer], (rootInstance) => {
+          const rootInstance = rootInstanceContainer.get();
```

Comment:
Here rootInstance is actually mutated though is recomputed right after createTransaction is complete.

Reply:
I don't understand the comment. Maybe you mistyped: "is actually mutated" should be "isn't actually mutated"?

Because `patchInstancesMutable` where we pass `rootInstance` doesn't mutate it.

Reply:
It does mutate though instancesIndex

Reply:
Ah, I think I get it: `instance.children = updates;` mutates `rootInstance`

Reply:
Really hard to understand what mutates what in this code:

```js
const rootInstance = rootInstanceContainer.get();
store.createTransaction([instancesStore], (instances) => {
            const { instancesById } = createInstancesIndex(rootInstance);
            const instance = instancesById.get(instanceId);
            if (instance) {
              instance.children = updates;
            }
            patchInstancesMutable(rootInstance, instances);
          });
```